### DISCUSSION TITLE
jQuery Core Inquiry

### DISCUSSION THREAD

**Discussion by jared-leddy** [jquery/jquery#5022] [Q&A]
### Description ###

I'm trying to build my own "lite" version of jQuery using only a few methods. Some examples are below.

$() // selector
.addClass
.removeClass
.attr
.setAttr
.text
.val
.ajax
.css

I'd like to create my own custom library of jQuery uses from the [core source code](https://code.jquery.com/jquery-3.6.0.js), but there is alot happening there. 

Is there any references that show what the "core" of jQuery requires? Something where I could take the "core" + "addClass" to give me just that .addClass() functionality.

**Comment by mgol** [comment]
> Is there any references that show what the "core" of jQuery requires?

This is a very open-ended question and I don't really know how to reply to this. Perhaps you can start by reading:
https://github.com/jquery/jquery#modules
and trying to build as small a custom jQuery build as possible and then see where you'd like to go from there.
